Tools/LyTestTools/ly_test_tools/cli/codeowners_hint.py

```diff
@@ -0,0 +1,232 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Tools for inspecting GitHub CODEOWNERS files, with a commandline interface
+"""
+
+import argparse
+import logging
+import os
+import pathlib
+import re
+import sys
+
+logger = logging.getLogger(__name__)
+
+_DEFAULT_CODEOWNER_ALIAS = "https://www.o3de.org/community/"
+_GITHUB_CODEOWNERS_BYTE_LIMIT = 3 * 1024 * 1024  # 3MB
+
+
+def get_codeowners(target_path: str) -> (str, str, str):
+    """
+    Finds ownership information matching the target filesystem path from a CODEOWNERS file found in its GitHub repo
+    :param target_path: path to match in a GitHub CODEOWNERS file, which will be discovered inside its repo
+    :return: tuple of (matched_path_entry, matched_owner_aliases, found_codeowners_path) which are empty when missing
+    """
+    codeowners_path = find_github_codeowners(target_path)
+    matched_path, owner_aliases = get_codeowners_from(target_path, codeowners_path)
+    return matched_path, owner_aliases, codeowners_path
+
+
+def find_github_codeowners(target_path: str) -> str:
```

Comment:
Why not have this functions accept a `pathlib.PurePath`?

Reply:
I see no reason not to, given the ones you've provided below. Will update!